### PR TITLE
[RHCLOUD-36365] Add metrics endpoint for worker

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-36365

## Description of Intent of Change(s)
Exposing metrics from worker pods
Related app-interface change: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/125038

## Local Testing
I added a counter in the cache status endpoint. 
Run podman compose build --no-cache rbac-worker; 
And run podman compose up -d rbac-worker
And call 0.0.0.0:6543 within the container, I can see the counter increased

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by alechenninger** [comment]
Looks like you have an extra file committed but otherwise LGTM! (rbac/celerybeat-schedule.db)

File: rbac/rbac/celery.py [hunk 1]
```diff
@@ -63,3 +66,12 @@
...
 app.autodiscover_tasks()
+
+
+@worker_ready.connect
+def start_metrics_server(sender=None, **kwargs):
+    """Start the metrics server."""
+    registry = CollectorRegistry()
+    multiprocess.MultiProcessCollector(registry)
+    generate_latest(registry)
+    start_http_server(LoadedConfig.metricsPort, addr="0.0.0.0", registry=registry)
```
Comment (lpichler) [inline rbac/rbac/celery.py:77]: can we add log message that server started ?

File: rbac/rbac/celery.py [hunk 2]
```diff
@@ -63,3 +66,12 @@
...
 # Load task modules from all registered Django app configs.
 app.autodiscover_tasks()
+
+
+@worker_ready.connect
+def start_metrics_server(sender=None, **kwargs):
+    """Start the metrics server."""
+    registry = CollectorRegistry()
+    multiprocess.MultiProcessCollector(registry)
+    generate_latest(registry)
```
Comment (lpichler) [inline rbac/rbac/celery.py:76]: how this work ? Is there any effect as return value is not used ? [ref](https://github.com/prometheus/client_python/blob/master/prometheus_client/exposition.py#L248)

Maybe `start_http_server` is enough to serve metrics ?

Comment (alechenninger) [reply rbac/rbac/celery.py:76]: Yeah good point, I think you're right.

**Review by alechenninger — APPROVED** [review]

**Pull request merged**